routes/courses.js

```diff
@@ -1,51 +1,133 @@
   @desc   Get courses with filter and paignation
   @access public
 */
-router.get('/', (req, res) => {
+router.get('/', async (req, res) => {
   try {
     const filters = req.query.filters || {};
-    const offset = parseInt(req.query.offset) || 0;
+    const page = parseInt(req.query.offset) || 0;
     const size = parseInt(req.query.size) || 10;
+    const offset = page * size;
 
-    const parsedFilters = JSON.parse(filters);
+    let parsedFilters = JSON.parse(filters);
+
+    parsedFilters = _.each(parsedFilters, (value, key) => {
+      if (key === 'year') {
+        let startYears = [];
+        let endYears = [];
+        _.each(value, year => {
+          const split = year.split('-');
+          startYears.push(Number(split[0]));
+          endYears.push(Number(split[1]));
+        });
+        parsedFilters.startYear = startYears;
+        parsedFilters.endYear = endYears;
+        delete parsedFilters.year;
+      }
+    });
 
     const filterKeys = Object.keys(parsedFilters);
-    const filtered = _.filter(MOCK_COURSES, course => {
-      for (let i = 0; i < filterKeys.length; i++) {
-        const filterKey = filterKeys[i];
-        const filterValue = parsedFilters[filterKey];
-        const courseValue = course[filterKey];
-
-        if (!filterValue.includes(courseValue)) {
-          return false;
+
+    let query = {};
+
+    if (filterKeys.length === 0) {
+      query = 'SELECT * FROM course';
+    } else {
+      let filterKey = filterKeys[0];
+      let filterValue = parsedFilters[filterKey];
+
+      filterKey = changeCase.snakeCase(filterKey);
+      filterValue = "'" + filterValue.join("','") + "'";
+
+      query = `SELECT * FROM course WHERE ${filterKey} IN (${filterValue})`;
+
+      if (filterKeys.length > 1) {
+        for (let i = 1; i < filterKeys.length; i++) {
+          filterKey = filterKeys[i];
+          filterValue = parsedFilters[filterKey];
+
+          filterKey = changeCase.snakeCase(filterKey);
+          filterValue = "'" + filterValue.join("','") + "'";
+
+          query = query + ` AND ${filterKey} IN (${filterValue})`;
         }
       }
-      return true;
+    }
+
+    let filtered = await client.query(query);
+    filtered = filtered.rows;
+
+    query = query + ` ORDER BY start_year DESC OFFSET ${offset} LIMIT ${size}`;
+
+    let sliced = await client.query(query);
+    sliced = sliced.rows;
+
+    sliced = _.each(sliced, course => {
+      const year = course.start_year + '-' + course.end_year;
+      course.year = year;
+      delete course.start_year;
+      delete course.end_year;
     });
 
-    const columnIds = Object.keys(MOCK_COURSES[0]);
-    const columnOptions = _.transform(
+    for (let i = 0; i < sliced.length; i++) {
+      let course = sliced[i];
+      course = _.each(course, (value, key) => {
+        key = changeCase.camelCase(key);
+        course[key] = value;
+      });
+    }
+
+    const sql = 'SELECT * FROM course ORDER BY start_year DESC';
```

Comment:
We might want to move this querying into its own endpoint to get the column options instead of having this as a part of this single GET request.